pro/src/new_components/OfferIndividualForm/UsefulInformations/Venue/__specs__/Venue.spec.tsx

```diff
@@ -0,0 +1,254 @@
+import userEvent from '@testing-library/user-event'
+import { Formik } from 'formik'
+import React from 'react'
+import * as yup from 'yup'
+
+import { TOffererName } from 'core/Offerers/types'
+import { TOfferIndividualVenue } from 'core/Venue/types'
+
+import { Venue, validationSchema } from '..'
+import { IVenueProps } from '../Venue'
+
+interface IInitialValues {
+  offererId: string
+  venueId: string
+}
+
+const renderVenue = ({
+  initialValues,
+  onSubmit = jest.fn(),
+  props,
+}: {
+  initialValues: IInitialValues
+  onSubmit: () => void
+  props: IVenueProps
+}) => {
+  const rtlReturns = render(
+    <Formik
+      initialValues={initialValues}
+      onSubmit={onSubmit}
+      validationSchema={yup.object().shape(validationSchema)}
+    >
+      <Venue {...props} />
+    </Formik>
+  )
+
+  const selectVenue = screen.getByLabelText('Lieu') as HTMLSelectElement
+  const selectOfferer = screen.getByLabelText('Structure') as HTMLSelectElement
+
+  return {
+    ...rtlReturns,
+    selectVenue,
+    selectOfferer,
+  }
+}
+
+describe('OfferIndividual section: venue', () => {
+  let initialValues: IInitialValues
+  let props: IVenueProps
+  let onSubmit = jest.fn()
+
+  beforeEach(() => {
+    const offererNames: TOffererName[] = [
+      {
+        id: 'AA',
+        name: 'Offerer AA',
+      },
+      {
+        id: 'BB',
+        name: 'Offerer BB',
+      },
+      {
+        id: 'CC',
+        name: 'Offerer CC',
+      },
+    ]
+
+    const venueList: TOfferIndividualVenue[] = [
+      {
+        id: 'AAAA',
+        name: 'Venue AAAA',
+        managingOffererId: 'AA',
+      },
+      {
+        id: 'BBAA',
+        name: 'Venue BBAA',
+        managingOffererId: 'BB',
+      },
+      {
+        id: 'CCAA',
+        name: 'Venue CCAA',
+        managingOffererId: 'CC',
+      },
+      {
+        id: 'CCBB',
+        name: 'Venue CCBB',
+        managingOffererId: 'CC',
+      },
+    ]
+
+    initialValues = {
+      offererId: '',
+      venueId: '',
+    }
+    props = {
+      offererNames,
+      venueList,
+    }
+  })
+
+  it('test initial render', () => {
+    const { selectVenue, selectOfferer } = renderVenue({
+      initialValues,
+      onSubmit,
+      props,
+    })
+
+    expect(selectOfferer).toBeInTheDocument()
+    const offererDefaultOption = screen.getByRole('option', {
+      name: 'Selectionner une structure',
+    }) as HTMLOptionElement
+    expect(offererDefaultOption.selected).toBeTruthy()
+    expect(selectOfferer.options.length).toBe(4)
+
+    expect(selectVenue).toBeInTheDocument()
+    const venueDefaultOption = screen.getByRole('option', {
+      name: 'Selectionner un lieu',
+    }) as HTMLOptionElement
+    expect(venueDefaultOption.selected).toBeTruthy()
+
+    // Select venue disabled when offererId is not defined
+    expect(selectVenue).toBeDisabled()
+    expect(selectVenue.options.length).toBe(1)
+  })
+
+  it('test initial render with a single value for offerer and venue', async () => {
```

Comment:
l'intitulé pourrait être plus explicite, par exemple `should atuomatically select the venue and offerer when there is only one in the list`

Reply:
j'ai l'impression aussi que le test pourrait être coupé en 2, ce sont 2 fonctionnalités différentes

Reply:
modification faite [ici](https://github.com/pass-culture/pass-culture-main/pull/2227/commits/06c9da4b6c9e7d662affb2c3c5a89bebc6febf6a)